awx/ui/src/components/LaunchButton/LaunchButton.js

```diff
@@ -60,6 +71,27 @@ function LaunchButton({ resource, children }) {
         setSurveyConfig(data);
       }
 
+      if (launch.ask_labels_on_launch) {
+        const {
+          data: { results },
+        } = await readLabels;
+
+        const allLabels = results.map((label) => ({
+          ...label,
+          isReadOnly: true,
+        }));
+
+        setLabels(allLabels);
+      }
+
+      if (launch.ask_instance_groups_on_launch) {
+        const {
+          data: { results },
+        } = await JobTemplatesAPI.readInstanceGroups(resource.id);
+
+        setInstanceGroups(results);
+      }
```

Comment:
These three requests (`readSurvey`, `readLabels` and `readInstanceGroups`) are made in serial. It would be more efficient to get both promises before awaiting the results (though it will require extra `if` blocks at the end before setting state vars).

Reply:
👍